{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "af71fdcc",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "27819fe3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene_Name</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2M-AS1</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A2MP1</th>\n",
       "      <th>A3GALT2</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>...</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "      <th>hsa-mir-1199</th>\n",
       "      <th>hsa-mir-125a</th>\n",
       "      <th>hsa-mir-335</th>\n",
       "      <th>hsa-mir-6080</th>\n",
       "      <th>hsa-mir-6723</th>\n",
       "      <th>hsa-mir-7162</th>\n",
       "      <th>hsa-mir-8072</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CGGA_1002</th>\n",
       "      <td>6.322946</td>\n",
       "      <td>2.166920</td>\n",
       "      <td>6.964970</td>\n",
       "      <td>-1.957072</td>\n",
       "      <td>-0.172562</td>\n",
       "      <td>-1.773320</td>\n",
       "      <td>-4.075655</td>\n",
       "      <td>2.285539</td>\n",
       "      <td>5.348583</td>\n",
       "      <td>5.278794</td>\n",
       "      <td>...</td>\n",
       "      <td>6.878821</td>\n",
       "      <td>2.833115</td>\n",
       "      <td>4.035353</td>\n",
       "      <td>5.909105</td>\n",
       "      <td>-3.816680</td>\n",
       "      <td>-4.796493</td>\n",
       "      <td>3.953017</td>\n",
       "      <td>7.573394</td>\n",
       "      <td>-9.965784</td>\n",
       "      <td>2.594931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1003</th>\n",
       "      <td>3.782405</td>\n",
       "      <td>0.573816</td>\n",
       "      <td>5.444542</td>\n",
       "      <td>-3.488867</td>\n",
       "      <td>1.158456</td>\n",
       "      <td>-4.323965</td>\n",
       "      <td>-2.665896</td>\n",
       "      <td>-0.247103</td>\n",
       "      <td>5.645185</td>\n",
       "      <td>2.176987</td>\n",
       "      <td>...</td>\n",
       "      <td>5.706760</td>\n",
       "      <td>4.750189</td>\n",
       "      <td>2.056585</td>\n",
       "      <td>3.933268</td>\n",
       "      <td>-1.968982</td>\n",
       "      <td>-3.656769</td>\n",
       "      <td>3.572968</td>\n",
       "      <td>8.620871</td>\n",
       "      <td>-4.323965</td>\n",
       "      <td>1.813170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1010</th>\n",
       "      <td>-1.710895</td>\n",
       "      <td>-3.982151</td>\n",
       "      <td>-0.634950</td>\n",
       "      <td>-6.532421</td>\n",
       "      <td>-6.532421</td>\n",
       "      <td>-8.240438</td>\n",
       "      <td>-9.965784</td>\n",
       "      <td>-6.695296</td>\n",
       "      <td>-0.793177</td>\n",
       "      <td>-3.378614</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.815354</td>\n",
       "      <td>-0.910249</td>\n",
       "      <td>-1.812429</td>\n",
       "      <td>-3.820627</td>\n",
       "      <td>-9.965784</td>\n",
       "      <td>-7.633351</td>\n",
       "      <td>-4.788841</td>\n",
       "      <td>-5.276715</td>\n",
       "      <td>-9.965784</td>\n",
       "      <td>-4.678208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1012</th>\n",
       "      <td>6.804116</td>\n",
       "      <td>2.002781</td>\n",
       "      <td>7.534239</td>\n",
       "      <td>-1.914960</td>\n",
       "      <td>0.844159</td>\n",
       "      <td>-2.688687</td>\n",
       "      <td>-1.976124</td>\n",
       "      <td>2.407788</td>\n",
       "      <td>5.582451</td>\n",
       "      <td>3.246323</td>\n",
       "      <td>...</td>\n",
       "      <td>6.778307</td>\n",
       "      <td>3.089121</td>\n",
       "      <td>4.140367</td>\n",
       "      <td>3.507378</td>\n",
       "      <td>-1.414061</td>\n",
       "      <td>-4.472973</td>\n",
       "      <td>3.051307</td>\n",
       "      <td>7.860614</td>\n",
       "      <td>-2.688687</td>\n",
       "      <td>1.049913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1014</th>\n",
       "      <td>6.730627</td>\n",
       "      <td>2.465862</td>\n",
       "      <td>6.432020</td>\n",
       "      <td>-1.647330</td>\n",
       "      <td>1.322725</td>\n",
       "      <td>-0.190777</td>\n",
       "      <td>-0.649591</td>\n",
       "      <td>1.881030</td>\n",
       "      <td>5.654234</td>\n",
       "      <td>4.779246</td>\n",
       "      <td>...</td>\n",
       "      <td>7.640308</td>\n",
       "      <td>2.913845</td>\n",
       "      <td>3.133410</td>\n",
       "      <td>4.657840</td>\n",
       "      <td>-2.451299</td>\n",
       "      <td>-3.853246</td>\n",
       "      <td>3.791191</td>\n",
       "      <td>7.611923</td>\n",
       "      <td>-4.112116</td>\n",
       "      <td>2.231094</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 23987 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Gene_Name      A1BG  A1BG-AS1       A2M   A2M-AS1     A2ML1     A2MP1  \\\n",
       "CGGA_1002  6.322946  2.166920  6.964970 -1.957072 -0.172562 -1.773320   \n",
       "CGGA_1003  3.782405  0.573816  5.444542 -3.488867  1.158456 -4.323965   \n",
       "CGGA_1010 -1.710895 -3.982151 -0.634950 -6.532421 -6.532421 -8.240438   \n",
       "CGGA_1012  6.804116  2.002781  7.534239 -1.914960  0.844159 -2.688687   \n",
       "CGGA_1014  6.730627  2.465862  6.432020 -1.647330  1.322725 -0.190777   \n",
       "\n",
       "Gene_Name   A3GALT2    A4GALT      AAAS      AACS  ...       ZYX     ZZEF1  \\\n",
       "CGGA_1002 -4.075655  2.285539  5.348583  5.278794  ...  6.878821  2.833115   \n",
       "CGGA_1003 -2.665896 -0.247103  5.645185  2.176987  ...  5.706760  4.750189   \n",
       "CGGA_1010 -9.965784 -6.695296 -0.793177 -3.378614  ... -1.815354 -0.910249   \n",
       "CGGA_1012 -1.976124  2.407788  5.582451  3.246323  ...  6.778307  3.089121   \n",
       "CGGA_1014 -0.649591  1.881030  5.654234  4.779246  ...  7.640308  2.913845   \n",
       "\n",
       "Gene_Name      ZZZ3  hsa-mir-1199  hsa-mir-125a  hsa-mir-335  hsa-mir-6080  \\\n",
       "CGGA_1002  4.035353      5.909105     -3.816680    -4.796493      3.953017   \n",
       "CGGA_1003  2.056585      3.933268     -1.968982    -3.656769      3.572968   \n",
       "CGGA_1010 -1.812429     -3.820627     -9.965784    -7.633351     -4.788841   \n",
       "CGGA_1012  4.140367      3.507378     -1.414061    -4.472973      3.051307   \n",
       "CGGA_1014  3.133410      4.657840     -2.451299    -3.853246      3.791191   \n",
       "\n",
       "Gene_Name  hsa-mir-6723  hsa-mir-7162  hsa-mir-8072  \n",
       "CGGA_1002      7.573394     -9.965784      2.594931  \n",
       "CGGA_1003      8.620871     -4.323965      1.813170  \n",
       "CGGA_1010     -5.276715     -9.965784     -4.678208  \n",
       "CGGA_1012      7.860614     -2.688687      1.049913  \n",
       "CGGA_1014      7.611923     -4.112116      2.231094  \n",
       "\n",
       "[5 rows x 23987 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CGGA_FPKM = pd.read_csv(\"./CGGA_Data/CGGA.mRNAseq_693.RSEM-genes.20200506.txt\", sep='\\t')\n",
    "CGGA_FPKM = CGGA_FPKM.set_index(\"Gene_Name\")\n",
    "CGGA_TPM = CGGA_FPKM.div(CGGA_FPKM.sum(axis=0), axis=1) * 1e6\n",
    "CGGA_TPM = np.log2(CGGA_TPM+0.001)\n",
    "CGGA_TPM = CGGA_TPM.T.copy()\n",
    "CGGA_TPM.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c5e365ac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene_Name</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>ABO</th>\n",
       "      <th>B3GALNT1</th>\n",
       "      <th>B3GALT1</th>\n",
       "      <th>B3GALT4</th>\n",
       "      <th>B3GALT5</th>\n",
       "      <th>B3GNT2</th>\n",
       "      <th>B3GNT5</th>\n",
       "      <th>B4GALNT1</th>\n",
       "      <th>B4GALT5</th>\n",
       "      <th>...</th>\n",
       "      <th>ST3GAL6</th>\n",
       "      <th>ST6GALNAC2</th>\n",
       "      <th>ST6GALNAC3</th>\n",
       "      <th>ST6GALNAC4</th>\n",
       "      <th>ST6GALNAC5</th>\n",
       "      <th>ST6GALNAC6</th>\n",
       "      <th>ST8SIA1</th>\n",
       "      <th>ST8SIA5</th>\n",
       "      <th>UGCG</th>\n",
       "      <th>UGT8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CGGA_1002</th>\n",
       "      <td>2.285539</td>\n",
       "      <td>-2.414404</td>\n",
       "      <td>3.698233</td>\n",
       "      <td>1.467149</td>\n",
       "      <td>2.676142</td>\n",
       "      <td>-1.463934</td>\n",
       "      <td>2.136633</td>\n",
       "      <td>1.622747</td>\n",
       "      <td>5.754396</td>\n",
       "      <td>2.854208</td>\n",
       "      <td>...</td>\n",
       "      <td>3.565270</td>\n",
       "      <td>-1.097098</td>\n",
       "      <td>1.207865</td>\n",
       "      <td>3.225497</td>\n",
       "      <td>3.339578</td>\n",
       "      <td>6.430430</td>\n",
       "      <td>2.702227</td>\n",
       "      <td>3.398200</td>\n",
       "      <td>2.646893</td>\n",
       "      <td>4.291344</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1003</th>\n",
       "      <td>-0.247103</td>\n",
       "      <td>-9.965784</td>\n",
       "      <td>0.880292</td>\n",
       "      <td>-0.058867</td>\n",
       "      <td>1.475185</td>\n",
       "      <td>-2.419415</td>\n",
       "      <td>1.080489</td>\n",
       "      <td>-0.149917</td>\n",
       "      <td>4.725320</td>\n",
       "      <td>2.417946</td>\n",
       "      <td>...</td>\n",
       "      <td>1.495356</td>\n",
       "      <td>-0.789941</td>\n",
       "      <td>-0.839607</td>\n",
       "      <td>1.895163</td>\n",
       "      <td>0.413466</td>\n",
       "      <td>4.907109</td>\n",
       "      <td>4.034953</td>\n",
       "      <td>0.094455</td>\n",
       "      <td>1.876059</td>\n",
       "      <td>2.436321</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1010</th>\n",
       "      <td>-6.695296</td>\n",
       "      <td>-9.965784</td>\n",
       "      <td>-7.207377</td>\n",
       "      <td>-5.560970</td>\n",
       "      <td>-5.727295</td>\n",
       "      <td>-7.335883</td>\n",
       "      <td>-7.476964</td>\n",
       "      <td>-4.615685</td>\n",
       "      <td>-0.290649</td>\n",
       "      <td>-2.257251</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.327610</td>\n",
       "      <td>-5.865971</td>\n",
       "      <td>-6.878927</td>\n",
       "      <td>-2.466621</td>\n",
       "      <td>-6.878927</td>\n",
       "      <td>-1.493856</td>\n",
       "      <td>-1.456846</td>\n",
       "      <td>-6.074407</td>\n",
       "      <td>-3.905349</td>\n",
       "      <td>-3.431487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1012</th>\n",
       "      <td>2.407788</td>\n",
       "      <td>-2.688687</td>\n",
       "      <td>2.318283</td>\n",
       "      <td>-0.199934</td>\n",
       "      <td>2.440336</td>\n",
       "      <td>-1.144628</td>\n",
       "      <td>2.906330</td>\n",
       "      <td>0.604101</td>\n",
       "      <td>3.787053</td>\n",
       "      <td>4.951028</td>\n",
       "      <td>...</td>\n",
       "      <td>3.850078</td>\n",
       "      <td>-1.594120</td>\n",
       "      <td>1.571928</td>\n",
       "      <td>4.319658</td>\n",
       "      <td>-0.858898</td>\n",
       "      <td>5.029946</td>\n",
       "      <td>3.879509</td>\n",
       "      <td>0.461459</td>\n",
       "      <td>2.639517</td>\n",
       "      <td>4.887530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1014</th>\n",
       "      <td>1.881030</td>\n",
       "      <td>-4.832545</td>\n",
       "      <td>3.338593</td>\n",
       "      <td>1.302925</td>\n",
       "      <td>2.234582</td>\n",
       "      <td>-0.502969</td>\n",
       "      <td>2.238062</td>\n",
       "      <td>2.370790</td>\n",
       "      <td>4.924001</td>\n",
       "      <td>2.994240</td>\n",
       "      <td>...</td>\n",
       "      <td>3.827448</td>\n",
       "      <td>1.793870</td>\n",
       "      <td>-0.100112</td>\n",
       "      <td>3.114602</td>\n",
       "      <td>2.571699</td>\n",
       "      <td>5.844024</td>\n",
       "      <td>3.129668</td>\n",
       "      <td>3.865068</td>\n",
       "      <td>2.444936</td>\n",
       "      <td>1.765215</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Gene_Name    A4GALT       ABO  B3GALNT1   B3GALT1   B3GALT4   B3GALT5  \\\n",
       "CGGA_1002  2.285539 -2.414404  3.698233  1.467149  2.676142 -1.463934   \n",
       "CGGA_1003 -0.247103 -9.965784  0.880292 -0.058867  1.475185 -2.419415   \n",
       "CGGA_1010 -6.695296 -9.965784 -7.207377 -5.560970 -5.727295 -7.335883   \n",
       "CGGA_1012  2.407788 -2.688687  2.318283 -0.199934  2.440336 -1.144628   \n",
       "CGGA_1014  1.881030 -4.832545  3.338593  1.302925  2.234582 -0.502969   \n",
       "\n",
       "Gene_Name    B3GNT2    B3GNT5  B4GALNT1   B4GALT5  ...   ST3GAL6  ST6GALNAC2  \\\n",
       "CGGA_1002  2.136633  1.622747  5.754396  2.854208  ...  3.565270   -1.097098   \n",
       "CGGA_1003  1.080489 -0.149917  4.725320  2.417946  ...  1.495356   -0.789941   \n",
       "CGGA_1010 -7.476964 -4.615685 -0.290649 -2.257251  ... -3.327610   -5.865971   \n",
       "CGGA_1012  2.906330  0.604101  3.787053  4.951028  ...  3.850078   -1.594120   \n",
       "CGGA_1014  2.238062  2.370790  4.924001  2.994240  ...  3.827448    1.793870   \n",
       "\n",
       "Gene_Name  ST6GALNAC3  ST6GALNAC4  ST6GALNAC5  ST6GALNAC6   ST8SIA1   ST8SIA5  \\\n",
       "CGGA_1002    1.207865    3.225497    3.339578    6.430430  2.702227  3.398200   \n",
       "CGGA_1003   -0.839607    1.895163    0.413466    4.907109  4.034953  0.094455   \n",
       "CGGA_1010   -6.878927   -2.466621   -6.878927   -1.493856 -1.456846 -6.074407   \n",
       "CGGA_1012    1.571928    4.319658   -0.858898    5.029946  3.879509  0.461459   \n",
       "CGGA_1014   -0.100112    3.114602    2.571699    5.844024  3.129668  3.865068   \n",
       "\n",
       "Gene_Name      UGCG      UGT8  \n",
       "CGGA_1002  2.646893  4.291344  \n",
       "CGGA_1003  1.876059  2.436321  \n",
       "CGGA_1010 -3.905349 -3.431487  \n",
       "CGGA_1012  2.639517  4.887530  \n",
       "CGGA_1014  2.444936  1.765215  \n",
       "\n",
       "[5 rows x 34 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GENE_LIST = ['A4GALT', 'ABO', 'B3GALNT1', 'B3GALT1', 'B3GALT4', 'B3GALT5', \n",
    "    'B3GNT2', 'B3GNT5', 'B4GALNT1', 'B4GALT5', 'B4GALT6', \n",
    "    'FUT1', 'FUT2', 'FUT3', 'FUT5', 'FUT6', 'FUT9', \n",
    "    'GAL3ST1', 'GCNT2', 'ST3GAL1', 'ST3GAL2', 'ST3GAL3', 'ST3GAL4', \n",
    "    'ST3GAL5', 'ST3GAL6', 'ST6GALNAC2', 'ST6GALNAC3', 'ST6GALNAC4', 'ST6GALNAC5', \n",
    "    'ST6GALNAC6', 'ST8SIA1', 'ST8SIA5', 'UGCG', 'UGT8']\n",
    "CGGA_TPM = CGGA_TPM[GENE_LIST]\n",
    "CGGA_TPM.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6acc4c4f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PRS_type</th>\n",
       "      <th>Histology</th>\n",
       "      <th>Grade</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age</th>\n",
       "      <th>OS</th>\n",
       "      <th>Censor (alive=0; dead=1)</th>\n",
       "      <th>Radio_status (treated=1;un-treated=0)</th>\n",
       "      <th>Chemo_status (TMZ treated=1;un-treated=0)</th>\n",
       "      <th>IDH_mutation_status</th>\n",
       "      <th>1p19q_codeletion_status</th>\n",
       "      <th>MGMTp_methylation_status</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CGGA_1002</th>\n",
       "      <td>Primary</td>\n",
       "      <td>AA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Female</td>\n",
       "      <td>43.0</td>\n",
       "      <td>305.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1003</th>\n",
       "      <td>Primary</td>\n",
       "      <td>O</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Female</td>\n",
       "      <td>47.0</td>\n",
       "      <td>3817.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>Codel</td>\n",
       "      <td>un-methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1010</th>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>45.0</td>\n",
       "      <td>246.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>un-methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1012</th>\n",
       "      <td>Recurrent</td>\n",
       "      <td>rO</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>45.0</td>\n",
       "      <td>3679.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>un-methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1014</th>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>42.0</td>\n",
       "      <td>263.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>un-methylated</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            PRS_type Histology    Grade  Gender   Age      OS  \\\n",
       "CGGA_ID                                                         \n",
       "CGGA_1002    Primary        AA  WHO III  Female  43.0   305.0   \n",
       "CGGA_1003    Primary         O   WHO II  Female  47.0  3817.0   \n",
       "CGGA_1010    Primary         A   WHO II    Male  45.0   246.0   \n",
       "CGGA_1012  Recurrent        rO   WHO II    Male  45.0  3679.0   \n",
       "CGGA_1014    Primary         A   WHO II    Male  42.0   263.0   \n",
       "\n",
       "           Censor (alive=0; dead=1)  Radio_status (treated=1;un-treated=0)  \\\n",
       "CGGA_ID                                                                      \n",
       "CGGA_1002                       1.0                                    1.0   \n",
       "CGGA_1003                       0.0                                    0.0   \n",
       "CGGA_1010                       1.0                                    1.0   \n",
       "CGGA_1012                       1.0                                    1.0   \n",
       "CGGA_1014                       1.0                                    0.0   \n",
       "\n",
       "           Chemo_status (TMZ treated=1;un-treated=0) IDH_mutation_status  \\\n",
       "CGGA_ID                                                                    \n",
       "CGGA_1002                                        1.0            Wildtype   \n",
       "CGGA_1003                                        1.0              Mutant   \n",
       "CGGA_1010                                        1.0              Mutant   \n",
       "CGGA_1012                                        1.0              Mutant   \n",
       "CGGA_1014                                        1.0            Wildtype   \n",
       "\n",
       "          1p19q_codeletion_status MGMTp_methylation_status  \n",
       "CGGA_ID                                                     \n",
       "CGGA_1002               Non-codel               methylated  \n",
       "CGGA_1003                   Codel            un-methylated  \n",
       "CGGA_1010                     NaN            un-methylated  \n",
       "CGGA_1012               Non-codel            un-methylated  \n",
       "CGGA_1014               Non-codel            un-methylated  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_data = pd.read_csv(\"./CGGA_Data/CGGA.mRNAseq_693_clinical.20200506.txt\", sep='\\t')\n",
    "clinical_data = clinical_data.set_index(\"CGGA_ID\")\n",
    "clinical_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7607c3f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "CGGA_TPM_Clinical = pd.merge(clinical_data, CGGA_TPM, left_index=True,right_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "cec729c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "LGG_CGGA = CGGA_TPM_Clinical[CGGA_TPM_Clinical[\"Grade\"].isin([\"WHO II\", \"WHO III\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0a8fef58",
   "metadata": {},
   "outputs": [],
   "source": [
    "LGG_CGGA = LGG_CGGA.drop(columns=['PRS_type', 'Histology', 'Grade', 'Gender', 'Age',\n",
    "                                    'Radio_status (treated=1;un-treated=0)',\n",
    "       'Chemo_status (TMZ treated=1;un-treated=0)', 'IDH_mutation_status',\n",
    "       '1p19q_codeletion_status', 'MGMTp_methylation_status'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "27bfada9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OS.time</th>\n",
       "      <th>OS</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>ABO</th>\n",
       "      <th>B3GALNT1</th>\n",
       "      <th>B3GALT1</th>\n",
       "      <th>B3GALT4</th>\n",
       "      <th>B3GALT5</th>\n",
       "      <th>B3GNT2</th>\n",
       "      <th>B3GNT5</th>\n",
       "      <th>...</th>\n",
       "      <th>ST3GAL6</th>\n",
       "      <th>ST6GALNAC2</th>\n",
       "      <th>ST6GALNAC3</th>\n",
       "      <th>ST6GALNAC4</th>\n",
       "      <th>ST6GALNAC5</th>\n",
       "      <th>ST6GALNAC6</th>\n",
       "      <th>ST8SIA1</th>\n",
       "      <th>ST8SIA5</th>\n",
       "      <th>UGCG</th>\n",
       "      <th>UGT8</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CGGA_1002</th>\n",
       "      <td>305.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.285539</td>\n",
       "      <td>-2.414404</td>\n",
       "      <td>3.698233</td>\n",
       "      <td>1.467149</td>\n",
       "      <td>2.676142</td>\n",
       "      <td>-1.463934</td>\n",
       "      <td>2.136633</td>\n",
       "      <td>1.622747</td>\n",
       "      <td>...</td>\n",
       "      <td>3.565270</td>\n",
       "      <td>-1.097098</td>\n",
       "      <td>1.207865</td>\n",
       "      <td>3.225497</td>\n",
       "      <td>3.339578</td>\n",
       "      <td>6.430430</td>\n",
       "      <td>2.702227</td>\n",
       "      <td>3.398200</td>\n",
       "      <td>2.646893</td>\n",
       "      <td>4.291344</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1003</th>\n",
       "      <td>3817.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.247103</td>\n",
       "      <td>-9.965784</td>\n",
       "      <td>0.880292</td>\n",
       "      <td>-0.058867</td>\n",
       "      <td>1.475185</td>\n",
       "      <td>-2.419415</td>\n",
       "      <td>1.080489</td>\n",
       "      <td>-0.149917</td>\n",
       "      <td>...</td>\n",
       "      <td>1.495356</td>\n",
       "      <td>-0.789941</td>\n",
       "      <td>-0.839607</td>\n",
       "      <td>1.895163</td>\n",
       "      <td>0.413466</td>\n",
       "      <td>4.907109</td>\n",
       "      <td>4.034953</td>\n",
       "      <td>0.094455</td>\n",
       "      <td>1.876059</td>\n",
       "      <td>2.436321</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1010</th>\n",
       "      <td>246.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-6.695296</td>\n",
       "      <td>-9.965784</td>\n",
       "      <td>-7.207377</td>\n",
       "      <td>-5.560970</td>\n",
       "      <td>-5.727295</td>\n",
       "      <td>-7.335883</td>\n",
       "      <td>-7.476964</td>\n",
       "      <td>-4.615685</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.327610</td>\n",
       "      <td>-5.865971</td>\n",
       "      <td>-6.878927</td>\n",
       "      <td>-2.466621</td>\n",
       "      <td>-6.878927</td>\n",
       "      <td>-1.493856</td>\n",
       "      <td>-1.456846</td>\n",
       "      <td>-6.074407</td>\n",
       "      <td>-3.905349</td>\n",
       "      <td>-3.431487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1012</th>\n",
       "      <td>3679.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.407788</td>\n",
       "      <td>-2.688687</td>\n",
       "      <td>2.318283</td>\n",
       "      <td>-0.199934</td>\n",
       "      <td>2.440336</td>\n",
       "      <td>-1.144628</td>\n",
       "      <td>2.906330</td>\n",
       "      <td>0.604101</td>\n",
       "      <td>...</td>\n",
       "      <td>3.850078</td>\n",
       "      <td>-1.594120</td>\n",
       "      <td>1.571928</td>\n",
       "      <td>4.319658</td>\n",
       "      <td>-0.858898</td>\n",
       "      <td>5.029946</td>\n",
       "      <td>3.879509</td>\n",
       "      <td>0.461459</td>\n",
       "      <td>2.639517</td>\n",
       "      <td>4.887530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1014</th>\n",
       "      <td>263.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.881030</td>\n",
       "      <td>-4.832545</td>\n",
       "      <td>3.338593</td>\n",
       "      <td>1.302925</td>\n",
       "      <td>2.234582</td>\n",
       "      <td>-0.502969</td>\n",
       "      <td>2.238062</td>\n",
       "      <td>2.370790</td>\n",
       "      <td>...</td>\n",
       "      <td>3.827448</td>\n",
       "      <td>1.793870</td>\n",
       "      <td>-0.100112</td>\n",
       "      <td>3.114602</td>\n",
       "      <td>2.571699</td>\n",
       "      <td>5.844024</td>\n",
       "      <td>3.129668</td>\n",
       "      <td>3.865068</td>\n",
       "      <td>2.444936</td>\n",
       "      <td>1.765215</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           OS.time   OS    A4GALT       ABO  B3GALNT1   B3GALT1   B3GALT4  \\\n",
       "CGGA_ID                                                                     \n",
       "CGGA_1002    305.0  1.0  2.285539 -2.414404  3.698233  1.467149  2.676142   \n",
       "CGGA_1003   3817.0  0.0 -0.247103 -9.965784  0.880292 -0.058867  1.475185   \n",
       "CGGA_1010    246.0  1.0 -6.695296 -9.965784 -7.207377 -5.560970 -5.727295   \n",
       "CGGA_1012   3679.0  1.0  2.407788 -2.688687  2.318283 -0.199934  2.440336   \n",
       "CGGA_1014    263.0  1.0  1.881030 -4.832545  3.338593  1.302925  2.234582   \n",
       "\n",
       "            B3GALT5    B3GNT2    B3GNT5  ...   ST3GAL6  ST6GALNAC2  \\\n",
       "CGGA_ID                                  ...                         \n",
       "CGGA_1002 -1.463934  2.136633  1.622747  ...  3.565270   -1.097098   \n",
       "CGGA_1003 -2.419415  1.080489 -0.149917  ...  1.495356   -0.789941   \n",
       "CGGA_1010 -7.335883 -7.476964 -4.615685  ... -3.327610   -5.865971   \n",
       "CGGA_1012 -1.144628  2.906330  0.604101  ...  3.850078   -1.594120   \n",
       "CGGA_1014 -0.502969  2.238062  2.370790  ...  3.827448    1.793870   \n",
       "\n",
       "           ST6GALNAC3  ST6GALNAC4  ST6GALNAC5  ST6GALNAC6   ST8SIA1   ST8SIA5  \\\n",
       "CGGA_ID                                                                         \n",
       "CGGA_1002    1.207865    3.225497    3.339578    6.430430  2.702227  3.398200   \n",
       "CGGA_1003   -0.839607    1.895163    0.413466    4.907109  4.034953  0.094455   \n",
       "CGGA_1010   -6.878927   -2.466621   -6.878927   -1.493856 -1.456846 -6.074407   \n",
       "CGGA_1012    1.571928    4.319658   -0.858898    5.029946  3.879509  0.461459   \n",
       "CGGA_1014   -0.100112    3.114602    2.571699    5.844024  3.129668  3.865068   \n",
       "\n",
       "               UGCG      UGT8  \n",
       "CGGA_ID                        \n",
       "CGGA_1002  2.646893  4.291344  \n",
       "CGGA_1003  1.876059  2.436321  \n",
       "CGGA_1010 -3.905349 -3.431487  \n",
       "CGGA_1012  2.639517  4.887530  \n",
       "CGGA_1014  2.444936  1.765215  \n",
       "\n",
       "[5 rows x 36 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LGG_CGGA.rename(columns={\"OS\":\"OS.time\",\"Censor (alive=0; dead=1)\":\"OS\"},inplace=True)\n",
    "LGG_CGGA.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "acc5ca1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "LGG_CGGA.to_csv(\"./CGGA_Data/CGGA_LGG_TPM.csv\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".conda",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
